com.civicrm.hrjobroles/CRM/Hrjobroles/BAO/HrJobRoles.php

```diff
@@ -21,7 +22,75 @@ public static function create($params) {
     return $instance;
   }
 
+  /**
+   * Get options for a given job roles field along with their database IDs.
+   *
+   * @param String $fieldName
+   *
+   * @return Array
+   */
+  public static function buildDbOptions($fieldName) {
+    $queryParam = array(1 => array($fieldName, 'String'));
+    $query = "SELECT t1.id, t1.label from civicrm_option_value t1
+              LEFT JOIN civicrm_option_group t2 on t1.option_group_id = t2.id
+              WHERE t2.name = %1";
+    $options = [];
+    $result = CRM_Core_DAO::executeQuery($query, $queryParam);
+    while ($result->fetch()) {
+      $options[] =  [ 'id'=>$result->id, 'label'=>strtolower($result->label) ];
+    }
+    return $options;
+  }
+
+  /**
+   * Check Contact if exist   .
+   *
+   * @param String $searchValue
+   * @param String $searchField
+   * @return Integer ( Contact ID or 0 if not exist)
+   */
+  public static function checkContact($searchValue, $searchField) {
+    $queryParam = array(1 => array($searchValue, 'String'));
+    $query = "SELECT id from civicrm_contact where ".$searchField." = %1";
+    $result = CRM_Core_DAO::executeQuery($query, $queryParam);
+    if ($result->fetch()) {
```

Comment:
Can this be changed to tertiary operator ? so maybe

`return $result->fetch() ? $result->id : 0;`

Again, your code is perfect here but this can save a few lines :)